packages/devtools_app/lib/src/inspector/inspector_breadcrumbs.dart

```diff
@@ -46,29 +49,27 @@ class InspectorBreadcrumbNavigator extends StatelessWidget {
     );
   }
 
-  List<_InspectorBreadcrumbData> _getBreadcrumbs(List<InspectorTreeRow> rows) {
-    final List<_InspectorBreadcrumbData> items = rows
-        .map((e) => _InspectorBreadcrumbData.wrap(e, e == rows.safeLast))
-        .toList();
-    if (items.length > 5) {
-      return []
-        ..add(items[0])
-        ..add(_InspectorBreadcrumbData.more())
-        ..addAll(items.sublist(items.length - 4, items.length));
+  List<_InspectorBreadcrumbData> _generateBreadcrumbs(
+    List<InspectorTreeRow> rows,
+  ) {
+    final List<_InspectorBreadcrumbData> items = rows.map((row) {
+      return _InspectorBreadcrumbData.wrap(
+        row: row,
+        isSelected: row == rows.safeLast,
+      );
+    }).toList();
+    List<_InspectorBreadcrumbData> breadcrumbs;
+    if (items.length > _maxNumberOfBreadcrumbs) {
+      breadcrumbs = [
+        items[0],
+        _InspectorBreadcrumbData.more(),
+        ...items.sublist(items.length - _maxNumberOfBreadcrumbs, items.length),
```

Comment:
shouldn't this be `items.length - _maxNumberOfBreadcrumbs - 1`? Otherwise this would happen
`[b1, b2, b3, b4, b5] --> b1 > more > b2, b3, b4, b5`

Reply:
I didn't like that `_maxNumberOfBreadcrumbs` meant max number of breadcrumbs excluding root so I've changed it to 5 (root + 4 widget breadcrumbs) with a comment. Now the logic looks like this:
```dart
[
    items[0],
    _InspectorBreadcrumbData.more(),
    ...items.sublist(items.length - _maxNumberOfBreadcrumbs + 1, items.length),
]
```

For 5 items it should take all items (root + allowed 4 at the end) without adding 'more' item. For 6+ items it should take first one and last 4 items, for example: `[b1, b2, b3, b4, b5, b6] -> b1 > more > b3, b4, b5, b6`